build_intermediary.py

```diff
@@ -50,8 +50,12 @@ def main(commands, threads, extension, additional_arg):
     parser.add_argument('-e', '--extension', type=str,
                         help='File extension type .i, .ll, .o etc',
                         default='.i')
-    parser.add_argument('-a', '--additional_arg', type=str,
-                        help='Any additional arg necessary to build extension type',
-                        default='-E')
+
     args = parser.parse_args()
-    main(args.commands, args.threads, args.extension, args.additional_arg)
+    arg_mapping = {
+        '.i': '-E',
+        '.s': '-S',
+        '.bc': '-emit-llvm',
+        '.ll': '-emit-llvm -S',
```

Comment:
the help text for `-e` mentions `.o`. Do we need an entry here for that?

Reply:
Maybe it's still useful to support `.o`? We'd be rebuilding most of the kernel without going through kbuild.